feat: landing section & main benefits section

simboonlong opened the pull request:
### User story
https://www.pivotaltracker.com/n/projects/2424361/stories/172105872

---

### PR updates

- add landing section with youtube overlay
- add benefits section, mainly textual
- translated overlay scss to emotion, removed overlay.module.scss
- removed about section

---

![Screenshot 2020-04-08 at 10 59 55 AM](https://user-images.githubusercontent.com/4774314/78740004-1dc22f80-7988-11ea-8c03-1e2ddd01fad5.png)

simboonlong committed:
feat: landing section & main benefits section
feat: add youtube overlay

Nebulis commented on src/components/HomePageContent/LandingSection.tsx:
```diff
@@ -0,0 +1,93 @@
+import React from "react";
```
once merge, could you update the picture of https://openattestation.com/

simboonlong replied:
yup sure

Nebulis commented on src/components/HomePageContent/MainBenefitsSection.tsx:
```diff
@@ -0,0 +1,91 @@
+              <img className="img-icon" src="/static/images/benefitsection/icon-trade.png" />
+            </div>
+            <div className="col">
+              <h5>Support for new service offerings through digitalisation and interoperability </h5>
```
eslint doesnt complain for the extra space at the end of the sentence ? :thinking:

Nebulis commented on src/components/Layout/Section.tsx:
```diff
@@ -0,0 +1,19 @@
+import React from "react";
+
+interface SectionProps {
+  columnClass?: string;
```
is it not used ? (by components using Section)

Nebulis commented on src/components/UI/Overlay/Overlay.tsx:
```diff
@@ -130,6 +298,16 @@ export const OverlayAddressBook = ({ addressBook = {}, ...props }: OverlayProps)
+  );
+};
+
+export const OverlayYoutube = ({ youtubeId, ...props }: OverlayProps) => {
```
I would suggest to remove youtubeId from OverlayProps and use the & operator from typescript or to extends OverlayProps:

1. 
```
export const OverlayYoutube = ({ youtubeId, ...props }: OverlayProps & {youtubeId: string})
```

2.
```
interface OverlayYoutubeProps extends OverlayProps{
  youtubeId: string;
}
```

simboonlong replied:
nice ok

Nebulis commented on src/components/UI/Overlay/Overlay.tsx:
```diff
@@ -63,11 +231,11 @@ export const OverlayAddressBook = ({ addressBook = {}, ...props }: OverlayProps)
   return (
-    <Overlay className={`${css.addressbook}`} {...props}>
-      <div className={css["overlay-actionsbar"]}>
+    <OverlayDefault className="addressbook" {...props}>
```
addressbook => address-book ?

Nebulis commented on src/components/UI/Overlay/Overlay.tsx:
```diff
@@ -1,38 +1,45 @@
 import React, { useState } from "react";
```
overlay doesn't lock the navigation ? (can still scroll)

simboonlong committed:
feat: escape to close overlay
feat: overlay to lock body
fix: extend props
fix: responsive tweaks
fix: shift verifier section to top fold, remove about section

Nebulis commented on src/common/hooks/useLockBodyScroll.ts:
```diff
@@ -0,0 +1,11 @@
+import { useLayoutEffect } from "react";
+
+export const useLockBodyScroll = () => {
```
do you have a magic trick to avoid the viewport to move when changing the property ?

My trick is to compute the size of the bar and add a padding right :see_no_evil:
